### PR TITLE
Fix docker build, add smoketests

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Add healthchecks to the server containers
- Add smoke tests to all the docker containers build in circleci
- Fix a bug where containers were being bootstrapped in a fashion did not reflect changes to requirements.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -276,6 +302,15 @@ commands:
             docker build -t vertexproject/synapse-axon:${CIRCLE_TAG} -f docker/images/axon/Dockerfile .
             docker build -t vertexproject/synapse-cortex:${CIRCLE_TAG} -f docker/images/cortex/Dockerfile .
             docker build -t vertexproject/synapse-cryotank:${CIRCLE_TAG} -f docker/images/cryotank/Dockerfile .
+      - run:
+          name: smoke test image
+          command: |
+            docker run --rm -it --entrypoint python vertexproject/synapse:${CIRCLE_TAG} -m synapse.servers.cortex --help
+            dstatus00=$?
```
Comment (mikemoritz) [inline .circleci/config.yml:309]: should this status be checked here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -297,6 +332,15 @@ commands:
             docker build -t vertexproject/synapse-axon:v2.x.x -f docker/images/axon/Dockerfile .
             docker build -t vertexproject/synapse-cortex:v2.x.x -f docker/images/cortex/Dockerfile .
             docker build -t vertexproject/synapse-cryotank:v2.x.x -f docker/images/cryotank/Dockerfile .
+      - run:
+          name: smoke test image
+          command: |
+            docker run --rm -it --entrypoint python vertexproject/synapse:v2.x.x -m synapse.servers.cortex --help
+            dstatus00=$?
```
Comment (mikemoritz) [inline .circleci/config.yml:339]: should this status be checked here?

Comment (vEpiphyte) [reply .circleci/config.yml:339]: yup!

File: .circleci/config.yml (hunk 1 above)
Comment (vEpiphyte) [reply .circleci/config.yml:309]: yup!

File: .circleci/config.yml (hunk 2 above)
Comment (vEpiphyte) [reply .circleci/config.yml:340]: fixed

File: .circleci/config.yml (hunk 1 above)
Comment (vEpiphyte) [reply .circleci/config.yml:309]: fixed

**Pull request merged**